File: csharp/Original/PublishingHouse.Application/Books/Commands/CreateDraftCommand.cs
using PublishingHouse.Authors;
using PublishingHouse.Books.Entities;
using PublishingHouse.Core.ValueObjects;

namespace PublishingHouse.Application.Books.Commands;

public record CreateDraftCommand(
    BookId BookId,
    Title Title,
    AuthorIdOrData Author,
    PublisherId PublisherId,
    PositiveInt Edition,
    Genre? Genre
);

File: java/src/main/java/io/eventdriven/slimdownaggregates/original/application/books/commands/CreateDraftCommand.java
package io.eventdriven.slimdownaggregates.original.application.books.commands;

import io.eventdriven.slimdownaggregates.original.domain.books.authors.AuthorIdOrData;
import io.eventdriven.slimdownaggregates.original.domain.books.entities.*;
import io.eventdriven.slimdownaggregates.original.infrastructure.valueobjects.PositiveInt;

public record CreateDraftCommand(
  BookId bookId,
  Title title,
  AuthorIdOrData author,
  PublisherId publisherId,
  PositiveInt edition,
  Genre genre
) {
}
